sds_data_manager/lambda_code/SDSCode/upload_api.py

```diff
@@ -69,14 +85,48 @@ def lambda_handler(event, context):
         return {
             "statusCode": 400,
             "body": json.dumps(
-                "No filename given for upload. "
-                "Please provide a filename "
+                "No filename given for the upload. Please provide a filename "
                 "in the path. Eg. /upload/path/to/file/filename.pkts"
             ),
         }
     # TODO: Handle other filetypes other than science files
     #      The current ScienceFilePath only accepts filenames, not the full path
     filename = os.path.basename(path_params)
+
+    # Supported SPICE extensions, must be a tuple for the endswith() check
+    spice_extensions = (".tsc", ".bsp", ".tf", ".tls", ".ck", ".tpc")
+    if filename.endswith(spice_extensions):
+        return spice_file_upload(filename)
+
+    # Upload for science files, which will catch other filetype errors
+    return science_file_upload(filename)
+
+
+def spice_file_upload(filename):
+    """Handle SPICE file uploads and place them in the proper location."""
+    filepath = f"imap/spice/{filename}"
+
+    # check if this SPICE file already exists, return a 409 if so
+    try:
+        S3_CLIENT.head_object(Bucket=BUCKET_NAME, Key=filepath)
+        # If the head_object operation succeeds, that means there
+        # is a file already at the specified path, so return a 409
+        return {
+            "statusCode": 409,
+            "body": json.dumps(f"{filepath} already exists."),
+            "headers": {
+                "Content-Type": "application/json",
+                "Access-Control-Allow-Origin": "*",
+            },
+        }
+    except botocore.exceptions.ClientError:
+        pass
```

Comment:
Did you mean to add some log here?

Reply:
No, this means a file wasn't found, but nothing else we should do. I added a quick comment for this pass that we just keep continuing.